sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Models/QueryAnalyzer.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.IoT.TimeSeriesInsights
+{
+    /// <summary>
+    /// The query analyzer allows a user to query for pages of data.
+    /// </summary>
+    public class QueryAnalyzer
+    {
+        private readonly QueryRequest _queryRequest;
+        private readonly QueryRestClient _queryClient;
+        private readonly string _storeType;
+        private CancellationToken _cancellationToken;
+
+        /// <summary>
+        /// Approximate progress of the query in percentage. It can be between 0 and 100. When the continuation token in the response is null,
+        /// the progress is expected to be 100.
+        /// </summary>
+        public double? Progress { get; internal set; }
+
+        /// <summary>
+        /// Initializes a new instance of the QueryResults class.
+        /// </summary>
+        /// <param name="queryClient">The query REST client that talks to TSI service.</param>
+        /// <param name="queryRequest">The query request payload.</param>
+        /// <param name="storeType">The store the query should be executed on.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        internal QueryAnalyzer(QueryRestClient queryClient, QueryRequest queryRequest, string storeType, CancellationToken cancellationToken)
+        {
+            _queryRequest = queryRequest;
+            _storeType = storeType;
+            _cancellationToken = cancellationToken;
+            _queryClient = queryClient;
+            Progress = 0;
+        }
+
+        /// <summary>
+        /// Get all of the <see cref="TimeSeriesPoint"/> asynchronously.
```

Comment:
Can reword as "The pageable list of ..."

Reply:
reworded to "`Get all of the <see cref="TimeSeriesPoint"/> in pages asynchronously.`" for consistency with other APIs